SDLTooSharp/Managed/Video/Surface/SDLSurface.cs

```diff
@@ -9,50 +10,33 @@ namespace SDLTooSharp.Managed.Video.Surface;
 /// <summary>
 /// Class for SDL Surface
 /// </summary>
-public partial class SDLSurface : ISurface, IDisposable
+[ExcludeFromCodeCoverage]
+public partial class SDLSurface : ISurface
 {
-    /// <summary>
-    /// Gets the actual pointer to an <see cref="SDLTooSharp.Bindings.SDL2.SDL.SDL_Surface"/> structure
-    /// </summary>
-    public IntPtr SurfacePtr { get; protected set; }
+    /// <inheritdoc cref="SurfacePtr"/>
+    public IntPtr SurfacePtr { get; init; }
 
     private Size _dimensions;
 
-    /// <summary>
-    /// Gets a value that indicates the dimensions of the surface
-    /// </summary>
+    /// <inheritdoc cref="Dimensions"/>
     public Size Dimensions => _dimensions;
 
     private int _pitch;
 
-    /// <summary>
-    /// Gets a value that indicates the pitch of the surface
-    /// (i.e. how many pixels there are per row)
-    /// </summary>
+    /// <inheritdoc cref="Pitch"/>
     public int Pitch => _pitch;
 
     private IntPtr _pixelPtr;
 
-    /// <summary>
-    /// Gets a value that indicates the pointer where the surface's pixels exist
-    /// </summary>
+    /// <inheritdoc cref="Pixels"/>
     public IntPtr Pixels => _pixelPtr;
```

Comment:
I particularly like how you utilized property inheritance for simplifying the interface implementation. We could further ensure that exceptions in critical methods like `Lock` and `Unlock` are documented to guide future developers.